src/stash/src/postgres.rs

```diff
@@ -106,14 +108,22 @@ impl Postgres {
         // Do the initial connection once here so we don't get stuck in transact's
         // retry loop if the url is bad.
         conn.connect().await?;
+
+        // Check if the fence table exists. Don't do this inside the txn because
+        // this conversion errors when the table isn't present, failing the
+        // transaction. There's a race condition here, but in the case that another
+        // thing is also trying to make a fence table, one of them should fail during
+        // commit. `to_regclass()` is an error-free solution to this, but Cockroach
+        // doesn't support it.
+        let fence_exists = conn
+            .client
+            .as_mut()
+            .unwrap()
+            .query_one("SELECT 'fence'::regclass", &[])
+            .await
+            .is_ok();
```

Comment:
Is it maybe easier to just query `pg_tables`?

```rust
conn.client.query_one(
    r#"SELECT EXISTS (
        SELECT FROM pg_tables
        WHERE schemaname = current_schema() and tablename = 'fence'
    )"#,
)
```

Reply:
Ah yeah this is better. Will change. I didn't think about how to deal with the search_path stuff, but it's current_schema!

Reply:
Yeah that puzzled me too at first!